app/src/main/java/com/amaze/filemanager/filesystem/FileUtil.java

```diff
@@ -334,11 +349,19 @@ public static final void writeUriToStorage(@NonNull final MainActivity mainActiv
                         }
                     }
                 }
-                return null;
+                return (retval.size() > 0) ? retval : null;
             }
 
             @Override
             public Void onPostExecute(Object result) {
+                if(result !=  null) {
+                    List<String> paths = (List<String>) result;
```

Comment:
Why are you casting `result`, can't it be of type `List<String>`.

Reply:
Implementation at `AppConfig.runInBackground()` is a bit messy that I couldn't find a way to add the type erasures back - for the time being.